gcc/rust/resolve/rust-ast-resolve.cc

```diff
@@ -195,6 +195,92 @@ ResolvePath::resolve_path (AST::PathInExpression *expr)
 		    expr->get_node_id (), expr->get_locus ());
 }
 
+void
+ResolvePath::resolve_path (AST::SimplePath *expr)
+{
+  // resolve root segment first then apply segments in turn
+  auto &segs = expr->get_segments ();
```

Comment:
It might be possible for SimplePaths to just do one big for loop instead of splitting this across two functions.